Platformio/HAL/HardwareAbstract.hpp

```diff
@@ -35,6 +40,11 @@ class HardwareAbstract {
   /// @param onBatteryStatusChangeHandler - Callable to be ran when batter status changes
   void onBatteryChange(std::function<void(batteryStatus)> onBatteryStatusChangeHandler);
 
+  void onStartWifiScan(std::function<void()> cb_func);
+  void onWifiScanDone(std::function<void(std::shared_ptr<std::vector<WifiInfo>>)> cb_func);
+  void notifyStartWifiScan();
+  void notifyWifiScanDone(std::shared_ptr<std::vector<WifiInfo>> info);
```

Comment:
with notification class you no longer really need these you can just use 

wifi_scan_start.notify(<vector to send out>)